src/Model/common.hpp

```diff
@@ -114,65 +122,121 @@ void seissol::model::getTransposedElasticCoefficientMatrix( seissol::model::Elas
+    R_inv(1,7) = neighbor.mu/(local.mu*(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho)));
+    R_inv(2,5) = 1/local.mu - neighbor.mu*sqrt(local.mu/local.rho)/(local.mu*local.mu*(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho)));
+    R_inv(2,8) = neighbor.mu/(local.mu*(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho)));
+    R_inv(3,4) = 1;
+    R_inv(4,0) = -local.lambda/(local.lambda + 2*local.mu) + (local.lambda*(neighbor.lambda + 2*neighbor.mu)/(local.lambda + 2*local.mu) - neighbor.lambda)*sqrt((local.lambda + 2*local.mu)/local.rho)/((local.lambda + 2*local.mu)*((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho)));
+    R_inv(4,1) = 1;
+    R_inv(4,6) = -(local.lambda*(neighbor.lambda + 2*neighbor.mu)/(local.lambda + 2*local.mu) - neighbor.lambda)/((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho));
+    R_inv(5,0) = -local.lambda/(local.lambda + 2*local.mu) + (local.lambda*(neighbor.lambda + 2*neighbor.mu)/(local.lambda + 2*local.mu) - neighbor.lambda)*sqrt((local.lambda + 2*local.mu)/local.rho)/((local.lambda + 2*local.mu)*((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho)));
+    R_inv(5,2) = 1;
+    R_inv(5,6) = -(local.lambda*(neighbor.lambda + 2*neighbor.mu)/(local.lambda + 2*local.mu) - neighbor.lambda)/((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho));
+    R_inv(6,5) = sqrt(local.mu/local.rho)/(local.mu*(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho)));
+    R_inv(6,8) = -1/(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho));
+    R_inv(7,3) = sqrt(local.mu/local.rho)/(local.mu*(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho)));
+    R_inv(7,7) = -1/(neighbor.mu*sqrt(local.mu/local.rho)/local.mu + sqrt(neighbor.mu/neighbor.rho));
+    R_inv(8,0) = sqrt((local.lambda + 2*local.mu)/local.rho)/((local.lambda + 2*local.mu)*((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho)));
+    R_inv(8,6) = -1/((neighbor.lambda + 2*neighbor.mu)*sqrt((local.lambda + 2*local.mu)/local.rho)/(local.lambda + 2*local.mu) + sqrt((neighbor.lambda + 2*neighbor.mu)/neighbor.rho));
+
+    Matrix99 chi = Matrix99::Zero();
+    chi(0,0) = 1.0;
+    chi(1,1) = 1.0;
+    chi(2,2) = 1.0;
+
+    auto godunov = ((R*chi)*R_inv).eval();
+
+    // QgodLocal = I - QgodNeighbor
+    for (unsigned i = 0; i < QgodLocal.shape(1); ++i) {
```

Comment:
Please write 9 or godunov.cols() or something like that for indices i and j. The code is going to fail whenever Qgodlocals shape is larger than 9 (e.g. for viscoelastic).